catroid/src/main/java/org/catrobat/catroid/stage/ShowTextActor.java

```diff
@@ -42,25 +43,31 @@
 import org.catrobat.catroid.formulaeditor.datacontainer.DataContainer;
 
 import java.util.List;
+import java.util.Locale;
 
 public class ShowTextActor extends Actor {
 
-	private final float textSize = 45;
+	private static final int DEFAULT_TEXT_SIZE = 45;
+	private float textSize;
 	private int xPosition;
 	private int yPosition;
+	private String color;
 	private UserVariable variableToShow;
 	private String variableNameToCompare;
 	private String variableValueWithoutDecimal;
 
 	private Sprite sprite;
 	private UserBrick userBrick;
 
-	public ShowTextActor(UserVariable userVariable, int xPosition, int yPosition, Sprite sprite, UserBrick userBrick) {
+	public ShowTextActor(UserVariable userVariable, int xPosition, int yPosition, float relativeSize, String color,
+			Sprite sprite, UserBrick userBrick) {
 		this.variableToShow = userVariable;
 		this.variableNameToCompare = variableToShow.getName();
 		this.variableValueWithoutDecimal = null;
 		this.xPosition = xPosition;
 		this.yPosition = yPosition;
+		this.textSize = DEFAULT_TEXT_SIZE * relativeSize;
```

Comment:
The new ShowTextAndColorBrick brick crashes Catroid for small and for very large size values. It also behaves strangely for negative size values (large negative size values move the shown value up the screen, for whatever reason). E.g, the app crashes when the brick is executed with a size value of 1%, or when the size is 100000%. 

It therefore is necessary to sanitize the size value to a legal range allowed by its implementation, and deal with all values above and below as having that highest, respectively lowest technically possible values. As an example, in case the technically feasible range would be from 5% up to 10000%: use 5 instead of 1 or -100, and use 10000 instead of 10001 and above.